File: apple/Sources/ReactNativeWatchOS/Modifiers/ProgressViewStyle.swift
import SwiftUI

/// SwiftUI `.progressViewStyle(_:)`. Maps the `style` string to a
/// progress-view style. Unknown values fall back to `.automatic`.
enum RNWProgressViewStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("progressViewStyle") { view, params, _ in
            switch params.string("style") {
            case "linear":
                return AnyView(view.progressViewStyle(.linear))
            case "circular":
                return AnyView(view.progressViewStyle(.circular))
            default:
                return AnyView(view.progressViewStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/progressViewStyle.ts
import { createModifier } from './createModifier';

export type ProgressViewStyle = 'automatic' | 'linear' | 'circular';

/// SwiftUI `.progressViewStyle(_:)`. Sets the visual style applied to
/// `ProgressView`s within the view. `'automatic'` defers to the platform
/// default.
export function progressViewStyle(style: ProgressViewStyle = 'automatic') {
  return createModifier('progressViewStyle', { style });
}
